src/model.cpp

```diff
@@ -560,14 +408,13 @@ bool doHasUnresolvedComponentImports(const ComponentPtr &component)
         if (!unresolvedImports) {
             // Check that the imported component can import all it needs from its model.
             ImportSourcePtr importedSource = component->importSource();
-            if (importedSource->hasModel()) {
-                ModelPtr importedModel = importedSource->model();
-                ComponentPtr importedComponent = importedModel->component(component->importReference());
-                if (importedComponent == nullptr) {
-                    unresolvedImports = true;
-                } else {
-                    unresolvedImports = doHasUnresolvedComponentImports(importedComponent);
-                }
+
+            ModelPtr importedModel = importedSource->model();
+            ComponentPtr importedComponent = importedModel->component(component->importReference());
```

Comment:
Could replace `ModelPtr` and `ComponentPtr` with `auto` (same for `ImportSourcePtr` above).

Reply:
db783e7f